generator/generator.md

```diff
@@ -0,0 +1,246 @@
+# Generator
+
+## Table of Contents
+
+- [How to Run the Script](#how-to-run-the-script)
+- [Generate Widgets](#generate-widgets)
+  - [Widgets Fields](#widgets-fields)
+  - [Widgets TypeScript](#widgets-typescript)
+  - [Widgets Example](#widgets-example)
+- [Generate Conditionals](#generate-conditionals)
+  - [Conditionals Fields](#conditionals-fields)
+  - [Conditionals TypeScript](#conditionals-typescript)
+- [Generate Choices](#generate-choices)
+  - [Choices Fields](#choices-fields)
+  - [Choices TypeScript](#choices-typescript)
+  - [Choices Example](#choices-example)
+- [Generate InputRange](#generate-inputrange)
+  - [InputRange Fields](#inputrange-fields)
+  - [InputRange TypeScript](#inputrange-typescript)
+  - [InputRange Example](#inputrange-example)
+- [Generate Libelles](#generate-libelles)
+  - [Libelles Fields](#libelles-fields)
+  - [Libelles YAML](#libelles-yaml)
+
+## How to Run the Script
+
+To run this script, follow these steps:
+
+1. Create an Excel file following this example : [Enquete_Teletravail_Generate_survey](https://polymtlca0-my.sharepoint.com/:x:/g/personal/samuel_duhaime-morissette_polymtl_ca/Ecp5FW8DQ5JFiYDEMCOb9PMBSUUbxdsaFq4_ncT5oxchWw?e=jsdIJt).
+2. Update the `.env` file with the necessary environment variables.
```

Comment:
Download the file should be separate from the generation. Ideally, this step would be optional and the survey can be generated from a local file

Reply:
Yes, I have a YAML file to activate or not the generateExcel.py. So we can already, generate the survey in local, or if the user prefer with the Excel Cloud. 

```YAML
excel:
    active_script: true
```